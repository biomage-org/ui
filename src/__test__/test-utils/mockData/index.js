import generateMockExperimentData from './mockExperimentData';
import generateMockProjects from './mockProjects';
import generateMockSamples from './mockSamples';
import generateMockExperiment from './mockExperiment';

// This file is a work in progress to generate mock data for the endpoints of the test.
// The generated data should be usable as response data to be sent via the mocked API in mockAPI.js
// The module should also expose the generated data that can be imported and used easily in tests
// A ticket has been created to give more thought into this https://biomage.atlassian.net/browse/BIOMAGE-1553

const responseData = {};

const projects = generateMockProjects(2);
responseData.projects = projects;

const projectWithSamples = responseData.projects[0];

responseData.samples = [generateMockSamples(
  projectWithSamples.uuid,
  projectWithSamples.experiments[0],
  3,
)];

const { samples } = responseData.samples[0];
const sampleIds = Object.keys(samples);

projectWithSamples.samples = sampleIds;

responseData.experiments = [generateMockExperiment(
  projectWithSamples.uuid,
  projectWithSamples.experiments[0],
  projectWithSamples.experiments[0],
  projectWithSamples.samples,
)];

responseData.experimentData = generateMockExperimentData(
  projectWithSamples.uuid,
  projectWithSamples.experiments[0],
  projectWithSamples.experiments[0],
  projectWithSamples.samples,
);

export {
  projects,
  samples,
  responseData,
};
