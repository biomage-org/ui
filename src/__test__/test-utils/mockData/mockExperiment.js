import _ from 'lodash';
import fake from '__test__/test-utils/constants';

const projectExperimentTemplate = (projectId, experimentId, experimentName, sampleIds) => (
  {
    lastViewed: fake.MOCK_DATETIME,
    apiVersion: '2.0.0-data-ingest-seurat-rds-automated',
    createdDate: fake.MOCK_DATETIME,
    meta: {
      pipeline: {
        stateMachineArn: 'arn:aws:states:eu-west-1:000000000000:stateMachine:biomage-qc-development-mock997586f1bae0a6d5c5beed65663ec151test',
        executionArn: 'arn:aws:states:eu-west-1:000000000000:execution:biomage-qc-development-mock997586f1bae0a6d5c5beed65663ec1519de4:422493f3-68a0-4ecb-8b17-0e8cd7a3test',
      },
      gem2s: {
        paramsHash: 'mock57f90e94eeaa82ee6fb7627110828f2etest',
        stateMachineArn: 'arn:aws:states:eu-west-1:000000000000:stateMachine:biomage-gem2s-development-mock997586f1bae0a6d5c5beed65663ec151test',
        executionArn: 'arn:aws:states:eu-west-1:000000000000:execution:biomage-gem2s-development-mock997586f1bae0a6d5c5beed65663ec1519de4:5f273383-f093-4112-95b0-d76db896test',
      },
      organism: null,
      type: '10x',
    },
    sampleIds,
    description: 'Analysis description',
    experimentId,
    projectId,
    experimentName,
  }
);

const generateMockProjectExperiments = (
  projectId,
  experimentId,
  experimentName,
  sampleIds = [],
  attrs = {},
) => _.merge(
  projectExperimentTemplate(
    projectId,
    experimentId,
    experimentName,
    sampleIds,
  ),
  attrs,
);

export default generateMockProjectExperiments;
