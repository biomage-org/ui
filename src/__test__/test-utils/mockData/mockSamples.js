import fake from '__test__/test-utils/constants';

const mockFileEntry = (fileName, sampleId) => ({
  valid: true,
  path: `${sampleId}/${fileName}`,
  upload: {
    amplifyPromise: null,
    progress: 100,
    status: 'uploaded',
  },
  name: fileName,
  compressed: true,
  lastModified: fake.MOCK_DATETIME,
  fileObject: {
    path: `${sampleId}/${fileName}`,
    size: 100,
  },
  errors: '',
});

const mockSampleTemplate = (projectId, sampleId, idx) => {
  const sampleTemplate = {
    metadata: {},
    createdDate: fake.LAST_MODIFIED,
    species: null,
    name: `Sample ${idx}`,
    files: {},
    lastModified: fake.LAST_MODIFIED,
    type: '10X Chromium',
    complete: false,
    error: false,
    uuid: `${sampleId}`,
    fileNames: [
      'features.tsv.gz',
      'barcodes.tsv.gz',
      'matrix.mtx.gz',
    ],
    projectUuid: `${projectId}`,
  };

  sampleTemplate.fileNames.forEach((fileName) => {
    sampleTemplate.files[fileName] = mockFileEntry(fileName, sampleId);
  });

  return sampleTemplate;
};

const generateMockSamples = (projectId, experimentId, numSamples = 3) => {
  const mockSample = {
    experimentId,
    projectUuid: projectId,
    samples: {},
  };

  for (let idx = 0; idx < numSamples; idx += 1) {
    const newSampleId = `${fake.SAMPLE_ID}-${idx}`;
    const mockSampleEntry = mockSampleTemplate(projectId, newSampleId, idx);

    mockSample.samples[newSampleId] = mockSampleEntry;
  }

  return mockSample;
};

export default generateMockSamples;
