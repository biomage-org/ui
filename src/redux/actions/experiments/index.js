import createExperiment from './createExperiment';
import updateExperiment from './updateExperiment';
import loadExperiments from './loadExperiments';
import loadExperiment from './loadExperiment';
import switchExperiment from './switchExperiment';
import reorderSamples from './reorderSamples';

export {
  createExperiment,
  updateExperiment,
  loadExperiments,
  loadExperiment,
  switchExperiment,
  reorderSamples,
};
