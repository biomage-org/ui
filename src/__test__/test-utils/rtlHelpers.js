import { fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

// dropFilesIntoDropzone takes an input element and simulates droping files into it
// inputElement: this needs to be the actual input element not the dropzone itself, e.g.:
// * <input data-testid='drop-input'/> => screen.getByTestId('drop-input');
// files should be a list of files, e.g.:
// * new File(['content'], 'file.txt', { type: 'text/plain' });
const dropFilesIntoDropzone = async (inputElement, files) => {
  await act(async () => {
    fireEvent.drop(inputElement, {
      target: { files },
    });
  });
};

export {
  // eslint-disable-next-line import/prefer-default-export
  dropFilesIntoDropzone,
};
