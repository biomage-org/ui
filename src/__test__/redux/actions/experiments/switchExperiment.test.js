import { switchExperiment, loadExperiments } from 'redux/actions/experiments';
import _ from 'lodash';
import mockAPI, {
  generateDefaultMockAPIResponses,

} from '__test__/test-utils/mockAPI';
import { makeStore } from 'redux/store';
import { loadProjects, setActiveProject } from 'redux/actions/projects';
import { projects } from '__test__/test-utils/mockData';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';

let store = null;

const projectWithSamples = projects.find((project) => project.samples.length > 0);
const projectWithoutSamples = projects.find((project) => project.samples.length === 0);

const experimentWithSamplesId = projectWithSamples.experiments[0];
const projectWithSamplesId = projectWithSamples.uuid;

const experimentWithoutSamplesId = projectWithoutSamples.experiments[0];
const projectWithoutSamplesId = projectWithoutSamples.uuid;

enableFetchMocks();

const mockAPIResponses = _.merge(
  generateDefaultMockAPIResponses(experimentWithSamplesId, projectWithSamplesId),
  generateDefaultMockAPIResponses(experimentWithoutSamplesId, projectWithoutSamplesId),
);

describe('switch experiment ', () => {
  beforeEach(async () => {
    fetchMock.resetMocks();
    fetchMock.doMockIf();
    fetchMock.mockIf(/.*/, mockAPI(mockAPIResponses));

    store = makeStore();
    await store.dispatch(loadProjects());
    await store.dispatch(loadExperiments(projectWithSamplesId));
    await store.dispatch(loadExperiments(projectWithoutSamplesId));
    await store.dispatch(setActiveProject(projectWithoutSamplesId));
  });

  it('switches the experiment to its initial values', async () => {
    await store.dispatch(switchExperiment(experimentWithSamplesId));

    expect(store.getState()).toMatchSnapshot();
  });

  it('Then updates the experiment info', async () => {
    await store.dispatch(switchExperiment(experimentWithSamplesId));

    expect(store.getState()).toMatchSnapshot();
  });
});
