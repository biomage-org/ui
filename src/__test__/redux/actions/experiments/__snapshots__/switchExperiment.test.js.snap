// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch experiment  Then updates the experiment info 1`] = `
Object {
  "backendStatus": Object {},
  "cellInfo": Object {
    "focus": Object {
      "key": "louvain",
      "store": "cellSets",
    },
    "groupedTrack": "louvain",
    "selectedTracks": Array [
      "louvain",
    ],
  },
  "cellMeta": Object {
    "doubletScores": Object {
      "data": Array [],
      "error": false,
      "loading": true,
    },
    "mitochondrialContent": Object {
      "data": Array [],
      "error": false,
      "loading": true,
    },
  },
  "cellSets": Object {
    "error": false,
    "hidden": Set {},
    "hierarchy": Array [],
    "loading": true,
    "properties": Object {},
    "selected": Object {},
    "updatingClustering": false,
  },
  "componentConfig": Object {},
  "differentialExpression": Object {
    "comparison": Object {
      "advancedFilters": Array [],
      "group": Object {
        "between": Object {
          "basis": null,
          "cellSet": null,
          "compareWith": null,
        },
        "within": Object {
          "basis": null,
          "cellSet": null,
          "compareWith": null,
        },
      },
      "ordering": Object {
        "orderBy": "logFC",
        "orderDirection": "DESC",
      },
      "type": "within",
    },
    "properties": Object {
      "comparisonGroup": Object {},
      "comparisonType": null,
      "data": Array [],
      "error": false,
      "loading": false,
      "total": 0,
    },
  },
  "embeddings": Object {},
  "experimentSettings": Object {
    "info": Object {
      "experimentId": "testae48e318dab9a1bd0bexperiment-0",
      "experimentName": "testae48e318dab9a1bd0bexperiment-0",
      "sampleIds": Array [
        "test9188-d682-test-mock-cb6d644cmock-0",
        "test9188-d682-test-mock-cb6d644cmock-1",
        "test9188-d682-test-mock-cb6d644cmock-2",
      ],
    },
    "originalProcessing": Object {},
    "processing": Object {
      "meta": Object {
        "changedQCFilters": Set {},
        "completingStepError": false,
        "loading": false,
        "loadingSettingsError": false,
      },
    },
  },
  "experiments": Object {
    "ids": Array [
      "testae48e318dab9a1bd0bexperiment-0",
    ],
    "meta": Object {
      "error": false,
      "loading": false,
      "saving": false,
    },
    "testae48e318dab9a1bd0bexperiment-0": Object {
      "createdDate": "0000-00-00T00:00:00.000Z",
      "description": "Analysis description",
      "id": "testae48e318dab9a1bd0bexperiment-0",
      "lastViewed": "0000-00-00T00:00:00.000Z",
      "meta": Object {
        "gem2s": Object {
          "executionArn": "arn:aws:states:eu-west-1:000000000000:execution:biomage-gem2s-development-mock997586f1bae0a6d5c5beed65663ec1519de4:5f273383-f093-4112-95b0-d76db896test",
          "paramsHash": "mock57f90e94eeaa82ee6fb7627110828f2etest",
          "stateMachineArn": "arn:aws:states:eu-west-1:000000000000:stateMachine:biomage-gem2s-development-mock997586f1bae0a6d5c5beed65663ec151test",
        },
        "organism": null,
        "pipeline": Object {
          "executionArn": "arn:aws:states:eu-west-1:000000000000:execution:biomage-qc-development-mock997586f1bae0a6d5c5beed65663ec1519de4:422493f3-68a0-4ecb-8b17-0e8cd7a3test",
          "stateMachineArn": "arn:aws:states:eu-west-1:000000000000:stateMachine:biomage-qc-development-mock997586f1bae0a6d5c5beed65663ec151test",
        },
        "type": "10x",
      },
      "name": "testae48e318dab9a1bd0bexperiment-0",
      "notifyByEmail": undefined,
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "sampleIds": Array [
        "test9188-d682-test-mock-cb6d644cmock-0",
        "test9188-d682-test-mock-cb6d644cmock-1",
        "test9188-d682-test-mock-cb6d644cmock-2",
      ],
    },
  },
  "genes": Object {
    "expression": Object {
      "data": Object {},
      "error": false,
      "loading": Array [],
      "views": Object {},
    },
    "focused": undefined,
    "markers": Object {
      "error": false,
      "loading": false,
    },
    "properties": Object {
      "data": Object {},
      "loading": Array [],
      "views": Object {},
    },
    "selected": Array [],
  },
  "layout": Object {
    "panel": "Gene list",
    "windows": Object {
      "direction": "row",
      "first": Object {
        "direction": "column",
        "first": Object {
          "direction": "row",
          "first": "UMAP",
          "second": "Data Management",
          "splitPercentage": 60,
        },
        "second": "Heatmap",
        "splitPercentage": 50,
      },
      "second": "Genes",
      "splitPercentage": 70,
    },
  },
  "networkResources": Object {
    "environment": undefined,
  },
  "projects": Object {
    "ids": Array [
      "test24b6-8600-test-mock-63822d2fmock-0",
      "test24b6-8600-test-mock-63822d2fmock-1",
    ],
    "loading": false,
    "meta": Object {
      "activeProjectUuid": "test24b6-8600-test-mock-63822d2fmock-1",
    },
    "test24b6-8600-test-mock-63822d2fmock-0": Object {
      "createdDate": "0000-00-00T00:00:00.000Z",
      "description": "Mock project 0",
      "experiments": Array [
        "testae48e318dab9a1bd0bexperiment-0",
      ],
      "lastAnalyzed": "0000-00-00T00:00:00.000Z",
      "lastModified": "0000-00-00T00:00:00.000Z",
      "metadataKeys": Array [],
      "name": "Mock project-0",
      "samples": Array [
        "test9188-d682-test-mock-cb6d644cmock-0",
        "test9188-d682-test-mock-cb6d644cmock-1",
        "test9188-d682-test-mock-cb6d644cmock-2",
      ],
      "uuid": "test24b6-8600-test-mock-63822d2fmock-0",
    },
    "test24b6-8600-test-mock-63822d2fmock-1": Object {
      "createdDate": "0000-00-00T00:00:00.000Z",
      "description": "Mock project 1",
      "experiments": Array [
        "testae48e318dab9a1bd0bexperiment-1",
      ],
      "lastAnalyzed": "0000-00-00T00:00:00.000Z",
      "lastModified": "0000-00-00T00:00:00.000Z",
      "metadataKeys": Array [],
      "name": "Mock project-1",
      "samples": Array [],
      "uuid": "test24b6-8600-test-mock-63822d2fmock-1",
    },
  },
  "samples": Object {
    "meta": Object {
      "error": false,
      "loading": false,
      "saving": false,
    },
    "test9188-d682-test-mock-cb6d644cmock-0": Object {
      "complete": false,
      "error": false,
      "fileNames": Array [
        "features.tsv.gz",
        "barcodes.tsv.gz",
        "matrix.mtx.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-0/barcodes.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "barcodes.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-0/barcodes.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-0/features.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "features.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-0/features.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-0/matrix.mtx.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "matrix.mtx.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-0/matrix.mtx.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "metadata": Object {},
      "name": "Sample 0",
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "species": null,
      "type": "10X Chromium",
      "uuid": "test9188-d682-test-mock-cb6d644cmock-0",
    },
    "test9188-d682-test-mock-cb6d644cmock-1": Object {
      "complete": false,
      "error": false,
      "fileNames": Array [
        "features.tsv.gz",
        "barcodes.tsv.gz",
        "matrix.mtx.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-1/barcodes.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "barcodes.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-1/barcodes.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-1/features.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "features.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-1/features.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-1/matrix.mtx.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "matrix.mtx.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-1/matrix.mtx.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "metadata": Object {},
      "name": "Sample 1",
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "species": null,
      "type": "10X Chromium",
      "uuid": "test9188-d682-test-mock-cb6d644cmock-1",
    },
    "test9188-d682-test-mock-cb6d644cmock-2": Object {
      "complete": false,
      "error": false,
      "fileNames": Array [
        "features.tsv.gz",
        "barcodes.tsv.gz",
        "matrix.mtx.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-2/barcodes.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "barcodes.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-2/barcodes.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-2/features.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "features.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-2/features.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-2/matrix.mtx.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "matrix.mtx.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-2/matrix.mtx.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "metadata": Object {},
      "name": "Sample 2",
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "species": null,
      "type": "10X Chromium",
      "uuid": "test9188-d682-test-mock-cb6d644cmock-2",
    },
  },
}
`;

exports[`switch experiment  switches the experiment to its initial values 1`] = `
Object {
  "backendStatus": Object {},
  "cellInfo": Object {
    "focus": Object {
      "key": "louvain",
      "store": "cellSets",
    },
    "groupedTrack": "louvain",
    "selectedTracks": Array [
      "louvain",
    ],
  },
  "cellMeta": Object {
    "doubletScores": Object {
      "data": Array [],
      "error": false,
      "loading": true,
    },
    "mitochondrialContent": Object {
      "data": Array [],
      "error": false,
      "loading": true,
    },
  },
  "cellSets": Object {
    "error": false,
    "hidden": Set {},
    "hierarchy": Array [],
    "loading": true,
    "properties": Object {},
    "selected": Object {},
    "updatingClustering": false,
  },
  "componentConfig": Object {},
  "differentialExpression": Object {
    "comparison": Object {
      "advancedFilters": Array [],
      "group": Object {
        "between": Object {
          "basis": null,
          "cellSet": null,
          "compareWith": null,
        },
        "within": Object {
          "basis": null,
          "cellSet": null,
          "compareWith": null,
        },
      },
      "ordering": Object {
        "orderBy": "logFC",
        "orderDirection": "DESC",
      },
      "type": "within",
    },
    "properties": Object {
      "comparisonGroup": Object {},
      "comparisonType": null,
      "data": Array [],
      "error": false,
      "loading": false,
      "total": 0,
    },
  },
  "embeddings": Object {},
  "experimentSettings": Object {
    "info": Object {
      "experimentId": "testae48e318dab9a1bd0bexperiment-0",
      "experimentName": "testae48e318dab9a1bd0bexperiment-0",
      "sampleIds": Array [
        "test9188-d682-test-mock-cb6d644cmock-0",
        "test9188-d682-test-mock-cb6d644cmock-1",
        "test9188-d682-test-mock-cb6d644cmock-2",
      ],
    },
    "originalProcessing": Object {},
    "processing": Object {
      "meta": Object {
        "changedQCFilters": Set {},
        "completingStepError": false,
        "loading": false,
        "loadingSettingsError": false,
      },
    },
  },
  "experiments": Object {
    "ids": Array [
      "testae48e318dab9a1bd0bexperiment-0",
    ],
    "meta": Object {
      "error": false,
      "loading": false,
      "saving": false,
    },
    "testae48e318dab9a1bd0bexperiment-0": Object {
      "createdDate": "0000-00-00T00:00:00.000Z",
      "description": "Analysis description",
      "id": "testae48e318dab9a1bd0bexperiment-0",
      "lastViewed": "0000-00-00T00:00:00.000Z",
      "meta": Object {
        "gem2s": Object {
          "executionArn": "arn:aws:states:eu-west-1:000000000000:execution:biomage-gem2s-development-mock997586f1bae0a6d5c5beed65663ec1519de4:5f273383-f093-4112-95b0-d76db896test",
          "paramsHash": "mock57f90e94eeaa82ee6fb7627110828f2etest",
          "stateMachineArn": "arn:aws:states:eu-west-1:000000000000:stateMachine:biomage-gem2s-development-mock997586f1bae0a6d5c5beed65663ec151test",
        },
        "organism": null,
        "pipeline": Object {
          "executionArn": "arn:aws:states:eu-west-1:000000000000:execution:biomage-qc-development-mock997586f1bae0a6d5c5beed65663ec1519de4:422493f3-68a0-4ecb-8b17-0e8cd7a3test",
          "stateMachineArn": "arn:aws:states:eu-west-1:000000000000:stateMachine:biomage-qc-development-mock997586f1bae0a6d5c5beed65663ec151test",
        },
        "type": "10x",
      },
      "name": "testae48e318dab9a1bd0bexperiment-0",
      "notifyByEmail": undefined,
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "sampleIds": Array [
        "test9188-d682-test-mock-cb6d644cmock-0",
        "test9188-d682-test-mock-cb6d644cmock-1",
        "test9188-d682-test-mock-cb6d644cmock-2",
      ],
    },
  },
  "genes": Object {
    "expression": Object {
      "data": Object {},
      "error": false,
      "loading": Array [],
      "views": Object {},
    },
    "focused": undefined,
    "markers": Object {
      "error": false,
      "loading": false,
    },
    "properties": Object {
      "data": Object {},
      "loading": Array [],
      "views": Object {},
    },
    "selected": Array [],
  },
  "layout": Object {
    "panel": "Gene list",
    "windows": Object {
      "direction": "row",
      "first": Object {
        "direction": "column",
        "first": Object {
          "direction": "row",
          "first": "UMAP",
          "second": "Data Management",
          "splitPercentage": 60,
        },
        "second": "Heatmap",
        "splitPercentage": 50,
      },
      "second": "Genes",
      "splitPercentage": 70,
    },
  },
  "networkResources": Object {
    "environment": undefined,
  },
  "projects": Object {
    "ids": Array [
      "test24b6-8600-test-mock-63822d2fmock-0",
      "test24b6-8600-test-mock-63822d2fmock-1",
    ],
    "loading": false,
    "meta": Object {
      "activeProjectUuid": "test24b6-8600-test-mock-63822d2fmock-1",
    },
    "test24b6-8600-test-mock-63822d2fmock-0": Object {
      "createdDate": "0000-00-00T00:00:00.000Z",
      "description": "Mock project 0",
      "experiments": Array [
        "testae48e318dab9a1bd0bexperiment-0",
      ],
      "lastAnalyzed": "0000-00-00T00:00:00.000Z",
      "lastModified": "0000-00-00T00:00:00.000Z",
      "metadataKeys": Array [],
      "name": "Mock project-0",
      "samples": Array [
        "test9188-d682-test-mock-cb6d644cmock-0",
        "test9188-d682-test-mock-cb6d644cmock-1",
        "test9188-d682-test-mock-cb6d644cmock-2",
      ],
      "uuid": "test24b6-8600-test-mock-63822d2fmock-0",
    },
    "test24b6-8600-test-mock-63822d2fmock-1": Object {
      "createdDate": "0000-00-00T00:00:00.000Z",
      "description": "Mock project 1",
      "experiments": Array [
        "testae48e318dab9a1bd0bexperiment-1",
      ],
      "lastAnalyzed": "0000-00-00T00:00:00.000Z",
      "lastModified": "0000-00-00T00:00:00.000Z",
      "metadataKeys": Array [],
      "name": "Mock project-1",
      "samples": Array [],
      "uuid": "test24b6-8600-test-mock-63822d2fmock-1",
    },
  },
  "samples": Object {
    "meta": Object {
      "error": false,
      "loading": false,
      "saving": false,
    },
    "test9188-d682-test-mock-cb6d644cmock-0": Object {
      "complete": false,
      "error": false,
      "fileNames": Array [
        "features.tsv.gz",
        "barcodes.tsv.gz",
        "matrix.mtx.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-0/barcodes.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "barcodes.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-0/barcodes.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-0/features.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "features.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-0/features.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-0/matrix.mtx.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "matrix.mtx.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-0/matrix.mtx.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "metadata": Object {},
      "name": "Sample 0",
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "species": null,
      "type": "10X Chromium",
      "uuid": "test9188-d682-test-mock-cb6d644cmock-0",
    },
    "test9188-d682-test-mock-cb6d644cmock-1": Object {
      "complete": false,
      "error": false,
      "fileNames": Array [
        "features.tsv.gz",
        "barcodes.tsv.gz",
        "matrix.mtx.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-1/barcodes.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "barcodes.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-1/barcodes.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-1/features.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "features.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-1/features.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-1/matrix.mtx.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "matrix.mtx.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-1/matrix.mtx.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "metadata": Object {},
      "name": "Sample 1",
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "species": null,
      "type": "10X Chromium",
      "uuid": "test9188-d682-test-mock-cb6d644cmock-1",
    },
    "test9188-d682-test-mock-cb6d644cmock-2": Object {
      "complete": false,
      "error": false,
      "fileNames": Array [
        "features.tsv.gz",
        "barcodes.tsv.gz",
        "matrix.mtx.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-2/barcodes.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "barcodes.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-2/barcodes.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-2/features.tsv.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "features.tsv.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-2/features.tsv.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "compressed": true,
          "errors": "",
          "fileObject": Object {
            "path": "test9188-d682-test-mock-cb6d644cmock-2/matrix.mtx.gz",
            "size": 100,
          },
          "lastModified": "0000-00-00T00:00:00.000Z",
          "name": "matrix.mtx.gz",
          "path": "test9188-d682-test-mock-cb6d644cmock-2/matrix.mtx.gz",
          "upload": Object {
            "amplifyPromise": null,
            "progress": 100,
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "metadata": Object {},
      "name": "Sample 2",
      "projectUuid": "test24b6-8600-test-mock-63822d2fmock-0",
      "species": null,
      "type": "10X Chromium",
      "uuid": "test9188-d682-test-mock-cb6d644cmock-2",
    },
  },
}
`;
